### PR TITLE
Sl 1904/spacing values

### PR DESCRIPTION
**Pull request by casserni**
With this update we will need to updated the spacing values in all relevant repos

- [x] ui/kit
- [x] studio https://github.com/stoplightio/studio/pull/167
- [x] json-schema-viewer https://github.com/stoplightio/json-schema-viewer/pull/9
- [x] json-schema-editor https://github.com/stoplightio/json-schema-editor/pull/12
- [x] tree-list (no values)
- [x] formtron https://github.com/stoplightio/formtron/pull/47
- [x] monaco (no values)
- [x] markdown-viewer https://github.com/stoplightio/markdown-viewer/pull/7

For reference old scale:
```
const spaceScale = {
  0: 0px,
  1: 4px,
  2: 8px,
  3: 16px,
  4: 32px,
  5: 64px,
  6: 128px,
};
```

new scale 
```
const space = {
  0: '0px',
  // increase by 2*0 = 1
  1: '1px',
  2: '2px',
  3: '3px',
  4: '4px',
  5: '5px',

  // increase by 2*1 = 2
  6: '6px',
  7: '8px',
  8: '10px',
  9: '12px',
  10: '14px',

  // increase by 2*2 = 4
  11: '18px',
  12: '22px',
  13: '26px',
  14: '30px',
  15: '34px',

  // increase by 2*3 = 6
  16: '40px',
  17: '46px',
  18: '52px',
  19: '60px',
  20: '66px',

  // increase by 2*4 = 8
  21: '78px',
  22: '86px',
  23: '94px',
  24: '102px',
  25: '110px',
};
```

pointers
```
old => new
  7 => 25
  6 => 25
  5 => 20
  4 => 15
  3 => 11
  2 => 7
  1 => 4
  0 => 0

```

### CODE REVIEW

File: src/__tests__/Flex.spec.tsx [hunk 1]
```diff
@@ -6,6 +6,10 @@ import { Box } from '../Box';
 import { Flex, IFlex } from '../Flex';
 
 describe('Flex component', () => {
+  beforeAll(async () => {
+    jest.spyOn(React, 'useContext').mockReturnValue({});
```
Comment (P0lip) [inline src/__tests__/Flex.spec.tsx]: Make sure to restore the spy.

```ts
let useContextSpy: jest.SpyInstance;

beforeAll(() => {
  useContextSpy = jest.spyOn(React, 'useContext').mockReturnValue({});
});

afterAll(() => {
  useContextSpy.mockRestore();
});
```

File: src/__tests__/Image.spec.tsx [hunk 2]
```diff
@@ -6,6 +6,10 @@ import { Box } from '../Box';
 import { Image } from '../Image';
 
 describe('Image component', () => {
+  beforeAll(async () => {
+    jest.spyOn(React, 'useContext').mockReturnValue({});
```
Comment (P0lip) [inline src/__tests__/Image.spec.tsx]: Ditto

File: src/__tests__/Dialog.spec.tsx [hunk 3]
```diff
@@ -32,6 +32,8 @@ describe('Dialog component', () => {
       return { Portal: fn };
     });
 
+    jest.spyOn(React, 'useContext').mockReturnValue({});
```
Comment (P0lip) [inline src/__tests__/Dialog.spec.tsx]: Ditto

File: src/styles.ts [hunk 4]
```diff
@@ -171,3 +176,46 @@ export interface IWhiteSpaceProps {
...
+/**
+ * SPACE
+ * we overwrite styled-system space function to have better space prop priority p > px > pr
+ */
+
+// @ts-ignore missing type
+export const space = ss.mapProps((props: ss.SpaceProps) => ({
+  ...props,
+  // need to pass theme to rule
+  theme: useTheme(),
```
Comment (P0lip) [inline src/styles.ts]: This looks very tricky.
Any other way we could avoid using a hook like this?

Comment (casserni) [reply src/styles.ts]: @P0lip I could call it in the box component then pass to the style prop, which is a little less confusing

Comment (P0lip) [reply src/styles.ts]: Yes, please. 
If someone adds an 'early' return, an exception will be thrown.

Comment (casserni) [reply src/styles.ts]: fixed

File: src/__tests__/Image.spec.tsx (hunk 2 above)
Comment (casserni) [reply src/__tests__/Image.spec.tsx]: fixed

File: src/__tests__/Flex.spec.tsx (hunk 1 above)
Comment (casserni) [reply src/__tests__/Flex.spec.tsx]: fixed

File: src/__tests__/Image.spec.tsx [hunk 5]
```diff
@@ -5,7 +5,16 @@ import * as React from 'react';
 import { Box } from '../Box';
 import { Image } from '../Image';
 
+let useContextSpy: jest.Mock<{}, any>;
```
Comment (P0lip) [inline src/__tests__/Image.spec.tsx]: Could be just `jest.SpyInstance`

File: src/__tests__/Dialog.spec.tsx (hunk 3 above)
Comment (casserni) [reply src/__tests__/Dialog.spec.tsx]: fixed

File: src/__tests__/Image.spec.tsx (hunk 5 above)
Comment (casserni) [reply src/__tests__/Image.spec.tsx]: done

**Pull request merged**